quinn-proto/src/token.rs

```diff
@@ -43,46 +51,44 @@ impl RetryToken {
                 .unwrap_or(0),
         );
 
-        let signature_pos = buf.len();
         match address.ip() {
             IpAddr::V4(x) => buf.put_slice(&x.octets()),
             IpAddr::V6(x) => buf.put_slice(&x.octets()),
         }
         buf.write(address.port());
         retry_src_cid.encode_long(&mut buf);
 
-        let signature = key.sign(&buf);
-        // No reason to actually encode the IP in the token, since we always have the remote addr for an incoming packet.
-        buf.truncate(signature_pos);
```

Comment:
We still shouldn't be transmitting the IP/port data, encrypted or not; it's just not needed. The added encryption means we can't use the cheeky virtual trailing data approach; instead, the IP and port should be passed in to the encryption/decryption process as additionally authenticated data (AAD).